seacrowd/sea_datasets/vistec_tp_th_21/vistec_tp_th_21.py

```diff
@@ -0,0 +1,174 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import re
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Licenses, Tasks
+
+_CITATION = """\
+@inproceedings{limkonchotiwat2021handling,
+  title={Handling Cross-and Out-of-Domain Samples in Thai Word Segmentation},
+  author={Limkonchotiwat, Peerat and Phatthiyaphaibun, Wannaphong and Sarwar, Raheem and Chuangsuwanich, Ekapol and Nutanong, Sarana},
+  booktitle={Findings of the Association for Computational Linguistics: ACL-IJCNLP 2021},
+  pages={1003--1016},
+  year={2021}
+}
+"""
+
+_DATASETNAME = "vistec_tp_th_21"
+
+_DESCRIPTION = """\
+The largest social media domain datasets for Thai text processing (word segmentation, 
+misspell correction and detection, and named-entity boundary) called "VISTEC-TP-TH-2021" or VISTEC-2021. 
+VISTEC corpus contains 49,997 sentences with 3.39M words where the collection was manually annotated by 
+linguists on four tasks, namely word segmentation, misspelling detection and correction, 
+and named entity recognition.
+"""
+
+_HOMEPAGE = "	https://github.com/mrpeerat/OSKut/tree/main/VISTEC-TP-TH-2021"
+
+
+_LANGUAGES = ["tha"]
+
+
+_LICENSE = Licenses.CC_BY_SA_3_0.value
+
+_LOCAL = False
+
+_URLS = {
+    "train": "https://raw.githubusercontent.com/mrpeerat/OSKut/main/VISTEC-TP-TH-2021/train/VISTEC-TP-TH-2021_train_proprocessed.txt",
+    "test": "https://raw.githubusercontent.com/mrpeerat/OSKut/main/VISTEC-TP-TH-2021/test/VISTEC-TP-TH-2021_test_proprocessed.txt",
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class VISTEC21(datasets.GeneratorBasedBuilder):
```

Comment:
Please add Dataset suffix to the class name

```suggestion
class VISTEC21Dataset(datasets.GeneratorBasedBuilder):
```